Board.java

```diff
@@ -80,44 +128,48 @@ public void setNextDirection(Game.Direction goHere) {
 		return board.clone();
 	}
 
-	public Creature get(Location l){
+	public Creature get( Location l ) {
 		return board[l.getX()][l.getY()];
 	}
 	
-	public void set(Location l, Creature c){
+	public void set( Location l, Creature c ) {
 		board[l.getX()][l.getY()] = c;
 		
-		if (c == Creature.Wall || c == Creature.Null || c == Creature.Point)
+		if ( isStaticCreature(c) )
 			return;
 		
-		c.setLocation(l);
+		MovingCreature Mc = (MovingCreature) c;
 		
-		if (c.getDirection() == null)
-			c.setDirection(Game.defaultDirection);
+		Mc.setLocation(l);
+		
+		if (Mc.getDirection() == null)
+			Mc.setDirection(Game.defaultDirection);
 	}
 	
-	public void limitToSpecificCreatures(Creature[] CreaturesForThisBoard){
-		for (int i = 0; i < board.length; i++){
-			for (int j = 0; j < board[0].length; j++){
-				if (!isCreatureAllowed( get(new Location(i, j) ), CreaturesForThisBoard ))
-					set( new Location(i, j), Creature.Null );
+	public void limitToSpecificCreatures( Creature[] CreaturesForThisBoard ) {
+		for ( int i = 0; i < board.length; i++ ) {
+			for ( int j = 0; j < board[0].length; j++ ) {
+				if ( !Arrays.stream( CreaturesForThisBoard ).anyMatch( get( new Location(i, j) ) :: equals ) ) 
+					set( new Location(i, j), Creature.NULL );
 			}
 		}
 	}
 	
-	public int getDimensions(){
-		if (board.length != board[0].length)
-			return Integer.parseInt( new String( board.length + "" + board[0].length ) );
-		
-		return board.length;
+	public int getDimensions( char dim ) {
+		return (dim == 'X') ? board.length : board[0].length;
 	}
 	
-	public int getGhostNum(){
+	public int getGhostNum() {
 		return this.ghostsNumber;
 	}
 	
-	private Creature[][] setBoard() throws FileNotFoundException{
-		Scanner sc = new Scanner(new BufferedReader(new FileReader("./Board.txt")));
+	public boolean isStaticCreature( Creature c ) {
+		return ( c == Creature.WALL || c == Creature.NULL || c == Creature.POINT ) ? true : false;
+	}
+	
+	private Creature[][] setBoard() throws FileNotFoundException {
+		// I didn't find the relative path for the file Board.txt so i used this

```

Comment:
You should try harder. think of it - you probably not the first developer to face such issue...